TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -154,17 +175,52 @@ public class Scanner
             Token nextToken = null;
-            while((nextToken = scanner.GetNextToken()) != null)
+            Token lastToken = null;
+            while ((nextToken = scanner.GetNextToken()) != null)
             {
                 if (nextToken.TokenType == TokenType.AlphanumericLiteral && (!nextToken.HasOpeningDelimiter || !nextToken.HasClosingDelimiter))
                 {
                     tokensLine.AddDiagnostic(MessageCode.SyntaxErrorInParser,
                         tokensLine.Indicator.StartIndex, tokensLine.Indicator.EndIndex, "Literal is not correctly delimited.");
                 }
+                bool nextTokenIncontinutaionLine = continuationPositions?.ContinuteBitPositions?.Get(nextToken.StartIndex) ?? false;
+                // Check if the previous AlphaNumericLiteral should be followed by a blank
+                bool previousAphaNumLitNeedSeparator = lastToken != null && lastToken.TokenType == TokenType.AlphanumericLiteral &&
+                    nextToken.TokenType != TokenType.SpaceSeparator && nextToken.TokenType != TokenType.PeriodSeparator &&
+                    ScannerUtils.IsIdentifierOrLiteral1(nextToken);
+                // Special check: not for a next token in a continuation line if the previous token was a literal string
+                if (previousAphaNumLitNeedSeparator)
+                {
+                    previousAphaNumLitNeedSeparator = !nextTokenIncontinutaionLine;
+                }
+
+                // Check if the next AlphaNumericLiteral should be preceed by a blank
+                bool nextAphaNumLitNeedSeparator = lastToken != null && lastToken.TokenType != TokenType.SpaceSeparator && lastToken.TokenType != TokenType.PeriodSeparator &&
+                    nextToken.TokenType == TokenType.AlphanumericLiteral &&
+                    ScannerUtils.IsIdentifierOrLiteral1(lastToken);
+                // Special check : not for a next token in a continuation line if the previous token was a literal string
+                if (nextAphaNumLitNeedSeparator && lastToken.TokenType == TokenType.AlphanumericLiteral)
+                {
+                    nextAphaNumLitNeedSeparator = !nextTokenIncontinutaionLine;
+                }
+                if (previousAphaNumLitNeedSeparator || nextAphaNumLitNeedSeparator)
+                {
+                    bool isOneQuoteFusion = lastToken.TokenType == TokenType.AlphanumericLiteral &&
+                        nextToken.TokenType == TokenType.AlphanumericLiteral &&
+                        lastToken.Text[lastToken.Text.Length - 1] == nextToken.Text[0];
+                    if (!isOneQuoteFusion)
+                    {   // Report only if the concatenation is not "" or ''
+                        tokensLine.AddDiagnostic(true, MessageCode.DotShouldBeFollowedBySpace,
+                            nextToken,
+                            nextToken.Text[0], Diagnostic.PredefArgColumnStart);
```

Comment:
Is it possible to compute final position here instead of using the predefined args mechanism ?

Reply:
In fact not because the token will be relocated later.